### PR TITLE
Add uml diagram

### PR DESCRIPTION
**Pull request by piesieomane**
In this Pull Request the following list below was implemented;

### CODE REVIEW

File: teacher.rb [hunk 1]
```diff
@@ -3,8 +3,8 @@
 class Teacher < Person
   attr_reader :specialization
 
-  def initialize(id, age, specialization, name = 'Unknown', parent_permission: true)
-    super(id, age, name, parent_permission)
+  def initialize(age, specialization, name = 'Unknown', parent_permission: true)
```
Comment (VICTOR-CHUKWU) [inline teacher.rb]: - [ ] There is an error while trying to create a teacher due to wrong number of arguments passed, kindly fix this
image

File: main.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
+require_relative 'App'
```
Comment (VICTOR-CHUKWU) [inline main.rb]: - [ ] kindly change `App` to `app` as you require relative

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @piesieomane  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 2 above)
Comment (piesieomane) [reply main.rb]: Thank you, this has been fixed now

File: teacher.rb (hunk 1 above)
Comment (piesieomane) [reply teacher.rb]: Thank you, this has been fixed now

**Review by dasileker — APPROVED** [review]
### **Hi @piesieomane,** 

### **Congratulations! 🎉**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 : 
 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**